foyer/forcefield.py

```diff
@@ -589,7 +633,7 @@ def createSystem(self, topology, atomtype=True, use_residue_map=True,
         # Execute scripts found in the XML files.
         for script in self._scripts:
             exec(script, locals())
-        return sys
+        return sys, data
```

Comment:
nit: Is there any other way to reach this data? Perhaps setting it as an attribute? This function is called `createSystem` so it's somewhat unintuitive that it would return other things as well.

Reply:
Something did feel a little off when I took this approach... `data` is only available from within `createSystem`, but yeah I could just throw a line like `self._system_data = data` or something like that before the return.